### PR TITLE
Constexpr hypot

### PR DESCRIPTION
**Pull request by mborland**

### CODE REVIEW

**Comment by mborland** [comment]
@NAThompson This should be good to go.

File: include/boost/math/ccmath/detail/swap.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace boost::math::ccmath::detail {
+
+template <typename T>
+inline constexpr void swap(T& x, T& y) noexcept
+{
+    T temp = x;
+    x = y;
+    y = temp;
+}
```
Comment (NAThompson) [inline include/boost/math/ccmath/detail/swap.hpp:17]: @mborland : Is there no constexpr `std::swap`?

**Comment by NAThompson** [comment]
@mborland : If you do find a `constexpr std::swap` just open up another PR. I figure you probably checked this . . .

File: include/boost/math/ccmath/detail/swap.hpp (hunk 1 above)
Comment (mborland) [reply include/boost/math/ccmath/detail/swap.hpp:17]: `std::swap` only became `constexpr` in [C++20.](https://en.cppreference.com/w/cpp/algorithm/swap)

**Pull request merged**